go-backend/db/mock.go

```diff
@@ -2,64 +2,131 @@ package db
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/bxcodec/faker/v3"
 	"github.com/stretchr/testify/mock"
 )
 
+// DBMockStore - test mock struct
 type DBMockStore struct {
 	mock.Mock
 }
 
+// GetRoleByID - test mock
+func (m *DBMockStore) GetRoleByID(ctx context.Context, id int) (role Role, err error) {
+	args := m.Called(ctx)
+	return args.Get(0).(Role), args.Error(1)
+}
+
+// GetRoleByName - test mock
+func (m *DBMockStore) GetRoleByName(ctx context.Context, name string) (role Role, err error) {
+	args := m.Called(ctx)
+	return args.Get(0).(Role), args.Error(1)
+}
+
+// GetOrganizationByDomainName - test mock
+func (m *DBMockStore) GetOrganizationByDomainName(ctx context.Context, domainName string) (organization Organization, err error) {
+	args := m.Called(ctx)
+	return args.Get(0).(Organization), args.Error(1)
+}
+
+// ListUsers - test mock
 func (m *DBMockStore) ListUsers(ctx context.Context) (users []User, err error) {
 	args := m.Called(ctx)
 	return args.Get(0).([]User), args.Error(1)
 }
 
+// CleanBlacklistedTokens - test mock
+func (m *DBMockStore) CleanBlacklistedTokens() (err error) {
+	return
+}
+
+// CreateUserBlacklistedToken - Mock for testing
+func (m *DBMockStore) CreateUserBlacklistedToken(ctx context.Context, blacklistedToken UserBlacklistedToken) (err error) {
+	return
+}
+
+// GetUserByEmail - Mock for retrieving a user. Just return a fake user object.
+func (m *DBMockStore) GetUserByEmail(ctx context.Context, email string) (user User, err error) {
+	user = User{}
```

Comment:
this is the wrong place to return a fake user. this method should only contain the mock code like:

```
args := m.Called(ctx, email)
return args.Get(0).(User), args.Error(1)
```

Then in test cases whenever you mock this call, you can tell the mock to return a fake user

dbMock.On("GetUserByEmail", mock.Anything, "test@test.com").Return(fake_user_object, nil)